src/test/java/com/teragrep/pth_03/tests/HeadSyntaxTests.java

```diff
@@ -135,4 +141,83 @@ void xpathTest5(String arg) throws Exception {
         // Check that 1 found
         assertEquals(1,nodesA.getLength());
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "head2",
+    })
+    void sortAfterHeadTest(String arg) { // parsing should continue after head command
+        ParserStructureTestingUtility pstu = new ParserStructureTestingUtility();
+        String fileName = "src/test/resources/antlr4/commands/head/" + arg + ".txt";
+        String integerExp = "/root/transformStatement/headTransformation/t_head_integerType/value";
+        String sortExp = "/root/transformStatement/transformStatement/sortTransformation/value";
+
+        NodeList integerNode = Assertions.assertDoesNotThrow(() -> (NodeList) pstu.xpathQueryFile(fileName, integerExp, true));
+        NodeList sortNode = Assertions.assertDoesNotThrow(() -> (NodeList) pstu.xpathQueryFile(fileName, sortExp, false));
+
+        // Check that 1 found
+        assertEquals(1, integerNode.getLength());
+        assertEquals(1, sortNode.getLength());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "head3",
+    })
+    void headWithEvalAndIntegerTest(String arg) { // should not parse, both integer and eval is not allowed
+        String fileName = "src/test/resources/antlr4/commands/head/" + arg + ".txt";
+        ParserSyntaxTestingUtility parserSyntaxTestingUtility
+                = new ParserSyntaxTestingUtility(fileName, false);
+        Assertions.assertThrows(Exception.class, () -> parserSyntaxTestingUtility.syntaxParseTest(arg));
```

Comment:
Can you use specific exception instead of the broadest one?

Reply:
Thanks, missed that one. Fixed now.